Drop unused db-devel dependency and conditional logic related to it

Conan-Kudo opened the pull request:
RPM can't even use the BDB library since RPM 4.17 anyway, so this
has just been hanging around for no reason at all.

davidebeatrici commented on rpm.spec:
```diff
@@ -514,12 +505,7 @@ autoreconf -i -f
 	--sharedstatedir=%{_var}/lib \
 	--with-archive \
 	--with-vendor=%{_real_vendor} \
-%ifarch %{oldarches}
-	--with-external-db \
 	--enable-bdb-ro \
-%else
-	--disable-bdb-ro \
-%endif
```
Shouldn't `--enable-bdb-ro` be removed?

Conan-Kudo replied:
No. It makes it possible to read bdb databases in chroots from other environments, which is quite useful for forensic purposes and such. bdb-ro doesn't use libdb at all.

berolinux replied:
The point of my commit was getting rid of bdb-ro on arches that never had a bdb backed rpm -- didn't relaize that bdb-ro somehow manages to access bdb databases without using libdb (probably through an internalized bdb version? Got to check on that...)

Conan-Kudo replied:
The BDB datafile format was reimplemented from scratch for bdb-ro.

davidebeatrici replied:
Ah, okay.

Conan-Kudo committed:
Drop unused db-devel dependency and conditional logic related to it

RPM can't even use the BDB library since RPM 4.17 anyway, so this
has just been hanging around for no reason at all.

Conan-Kudo commented:
I assume someone took care of this by now. I still can't commit to the repo for some reason. 🤷🏾‍♂️
